src/app/component/projects/pages/details/details.component.ts

```diff
@@ -82,6 +84,13 @@ export class DetailsComponent extends PageComponent implements OnInit {
 
         this.sitesApi.getProjectSites(params.projectId).subscribe({
           next: sites => {
+            sites.map(site => {
+              this.avgLat += site.customLatitude;
+              this.avgLong += site.customLongitude;
+            });
+            this.avgLat /= sites.length;
+            this.avgLong /= sites.length;
+
```

Comment:
Averaging of lat/long doesn't work like this.

But more so, why are we averaging?

Reply:
We need the average latitude and longitude for the `<agm-map>` component as it does not have an inbuilt averaging feature (as far as I tested, will look over again). I will look into a better implementation of this calculation.

Reply:
No, please do no implement such an algorithm.

But what is being averaged here? Are we averaging the site locations? Why?

Previous maps just showed all sites in the project details page - why not again?

Reply:
This is to average the site locations, its needed because you need to supply the `<agm-map>` component with a longitude and latitude to focus the map on. Adding pins has no affect on the zoom level or position of the map. Below is an image of the map that is generated. The pins are added using a for loop in the html which iterates over the sites and adds pins, snazzy-info-window creates the white background for the pin label, and the average latitude and longitude focus the map to where the pins are located.
![image](https://user-images.githubusercontent.com/3955116/67116872-a0d8a900-f1e1-11e9-8008-f557691f4e13.png)

Reply:
Newest commit found how to use the agm/core library to auto focus on the pins

Reply:
Fantastic 👍